### PR TITLE
Updated documentation for v0.2

### PR DESCRIPTION
**Pull request by HGSilveri**
This PR updates the documentation for the upcoming release of v0.2.

@sebgrijalva I would kindly ask you to look at the compiled website (from the checks) and see what you think, no need to review the code. The notable changes are in:
- The structure of the sections
- The new tutorials
- The extended text in the "Fundamental Concepts" section

Once the parametrized sequence tutorial is merged, I will add it to the docs too, under "Advanced features".

### CODE REVIEW

File: docs/source/review.rst [hunk 1]
```diff
@@ -18,9 +18,14 @@ principal quantum number. At the Rydberg state, the atoms present long
 lifetimes and large interaction strengths and are briefly called
 **Rydberg atoms**.
 
-**Pulser** is an open-source software package written in Python whose main
-objective is to provide easy-to-use libraries for designing and simulating
-pulse sequences that act in programmable Rydberg atom arrays.
+This pages page gives brief insight into the concepts behind **Pulser** and
```
Comment (sebgrijalva) [inline docs/source/review.rst:21]: Remove extra "pages"

File: docs/source/review.rst [hunk 2]
```diff
@@ -70,33 +75,51 @@ into the physics and try to understand the underlying **Hamiltonian** of our
...
-Quantum Hamiltonian
-*******************
+Driving two-level transitions
+******************************
 
-Let us call the initial state of the atoms their **ground state** and denote it
-by :math:`|g\rangle`. The incident laser field will then try to excite them to
-a Rydberg level, which we shall write as :math:`|r\rangle`. These two states
-can be thought of as the two levels of a quantum spin.
+Each channel is tuned such each of its pulses coherently drives a
```
Comment (sebgrijalva) [inline docs/source/review.rst:81]: should be "such that" each of its pulses

File: docs/source/review.rst [hunk 3]
```diff
@@ -137,23 +160,41 @@ In the analog simulation approach, the laser field acts on the entire array
...
-an excitation which will be blocked (or not) depending on the distance and
-current levels of the involved atoms.
+In opposition to the analog approach stands the digital approach, in which
+a system's state evolves through a series of discrete manipulations of its qubits'
+states, also known as quantum gates. This is the underlying approach in quantum
+circuits and it can be replicated on neutral-atom devices at the pulse-level. To
+this extent, the qubit states are encoded in two hyperfine ground states of the
+system, named `ground`, :math:`|g\rangle`, and `hyperfine`, :math:`|h\rangle`. In
+Pulser, these states form the ``digital`` basis, which is addressed by ``Raman``
+channels. In the digital approach, these channels are usual `Local`, meaning they
```
Comment (sebgrijalva) [inline docs/source/review.rst:182]: are "usually" `Local`

File: docs/source/review.rst [hunk 4]
```diff
@@ -137,23 +160,41 @@ In the analog simulation approach, the laser field acts on the entire array
...
+target individual qubits instead of the entire system.
+
+Since the Rydberg blockade effect is not present when atoms are in :math:`|g\rangle` or :math:`|h\rangle`,
+the dynamics of each qubit's state are just determined by the driving pulses,
+already illustrated above as rotations on the Block sphere. By tuning the parameters of a pulse,
```
Comment (sebgrijalva) [inline docs/source/review.rst:187]: on the Bloc**h** sphere

File: docs/source/review.rst [hunk 5]
```diff
@@ -137,23 +160,41 @@ In the analog simulation approach, the laser field acts on the entire array
...
+we can achieve any arbitrary single-qubit unitary, as is show in :doc:`phase_shifts_vz_gates`.
```
Comment (sebgrijalva) [inline docs/source/review.rst:188]: as is show**n**

**Review by sebgrijalva — COMMENTED** [review]
Here are some typos I found on the docs. I will proceed now with the Notebooks

**Review by sebgrijalva — APPROVED** [review]
Notebooks look good, thanks a lot for the efforts into Pulser!

**Comment by HGSilveri** [comment]
@sebgrijalva  Thanks a lot for the review, I'll fix all the typos right away!

**Pull request merged**